frontend/coprs_frontend/coprs/views/webhooks_ns/webhooks_general.py

```diff
@@ -79,6 +80,20 @@ def decorated_function(copr, **kwargs):
 
     return decorated_function
 
+def add_webhook_history_record(webhook_uuid, user_agent='Not Set', builds_initiated_via_hook=None):
+    """
+    This methods adds info of an intercepted webhook to webhook_history db
+    along with the initiated build number(s).
+    """
+    if builds_initiated_via_hook is None:
+        log.debug("No build initiated. Webhook not logged to db.")
+        return
+
+    #For each build that was initiated by the webhook insert a record in db.
+    for build in builds_initiated_via_hook:
+        webhookRecord = models.WebhookHistory(webhook_uuid=webhook_uuid,
+                                              user_agent=user_agent, build_id=build)
+        db.session.add(webhookRecord)
```

Comment:
I am really sorry, I wrote a detailed comment about this last week but when I looked here today, it somehow disappeared. It probably didn't submit properly because not even my teammates got a notification about it :-/

If I understand this function correctly, when a webhook submits 3 builds, it creates 3 rows in the `webhook_history` table, all of them containing the same information except for the referenced build? I would like to see the opposite behavior when only one `webhook_history` row is created regardless of how many builds it triggered. The `webhook_history` table wouldn't reference any builds, it would be the other way around and we would have something like `webhook_history_id` in the `build` table and the builds would reference the webhook that triggered them.

But we will probably have to add `copr_id` to the `webhooks_history` table so that we can easily show all webhooks for a given project. AFAIK one webhook cannot trigger builds from multiple projects at once, so this should be fine.

Reply:
No worries at all, @FrostyX 

For some reason I wrongly assumed I had to implement it such that I do not touch any existing tables. Don't ask. 😄 

So, new change implements it such that a webhook history is committed and a reference to it's `id` is added within a new column in `Build` table named `webhook_history_id`